plugin/__init__.py

```diff
@@ -1 +1,22 @@
-__version__ = (0, 10, 1)
+from .core.protocol import Notification
+from .core.protocol import Request
+from .core.protocol import Response
+from .core.protocol import WorkspaceFolder
+from .core.sessions import __version__
+from .core.sessions import Session
+from .core.types import ClientConfig
+from .core.types import LanguageConfig
+from .core.settings import read_client_config
+
+# This is the public API for LSP-* packages
+__all__ = [
+    '__version__',
+    'ClientConfig',
+    'LanguageConfig',
+    'Notification',
+    'read_client_config',
+    'Request',
+    'Response',
+    'Session',
+    'WorkspaceFolder',
+]
```

Comment:
If `NpmClientHandler` would override `__init__`, it would also need to have access to extra types that are arguments of `Session.__init__` (assuming that it would want to properly annotate types). And I guess that would be kinda bad because __init__ wouldn't be necessarily stable. So I think we might actually want public class that we would override, as you've noted in https://github.com/sublimelsp/lsp_utils/pull/8. That would make thing a lot more proper in terms of having a stable API.

Reply:
It can be avoided like this:

```python
class MySession(Session):

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        # do init stuff here
```